sig-and-wg/sig-core/proposals/modularization.md

```diff
@@ -0,0 +1,72 @@
+
+## Goal
+
+Allow users to add/remove/replace Kyma components in more flexible way during the installation and in runtime. This should be possible:
+
+- install Kyma lite on minikube (without monitoring, console and ui-api-layer, tracing, service catalog, and remote environments)
+- install Kyma without service catalog and brokers installed on GKE with already provided Service Catalog and GCP broker included
+- install Kyma as gardener addon Gardener with dependencies to Istio and Knative addons
+- add service broker to existing Kyma installation in runtime (e.g. azure broker)
+- add new remote environment (also new type of application) in runtime
+- add completely new Kyma component that integrates with UI in runtime (register microfrontend and custom resources)
+- use user provided S3 bucket instead of minio
+- install private modules (from private docker registry or private git repository)
+- install components from Kyma examples/showcases catalog easily (UI)
+
+## Current solution
+
+Current installation process is a mix of:
+
+- custom shell scripts, 
+- kubectl commands,
+- helm charts including one big, fat core chart, 
+- complex parametrization
+
+Installer is a kubernetes operator with hard coded logic executing all the components above. Problems with this solution:
+
+- default minikube installation is slow and require a lot of resources
+- shell scripts are difficult to manage (some of them have to support mac, linux, windows)
+- new installation parameters or dependencies require code changes
+
+## Proposal
+
+Flexible and modular installation of fine grained components with limited dependencies. The components installation and update should be more generic and declarative. This is the first list of ideas how to improve current solution:
+
+- parameters should not be hard coded - there should be a convention how to pass parameters through installation process to Kyma components
+- default values everywhere - the list of mandatory parameters should be as short as possible (the goal: only cluster domain is a mandatory parameter)
+- optional dependencies, e.g. Kyma ui-api-layer can work without custom resources introduced by service catalog or remote environment - still can support other areas.
+- dynamic UI (microfrontends) + back-end - adding new components extend views (but also extend backed functionality like supported graphQL queries)
+- no custom logic in installation process (shell scripts) - it required it should be replaced by the operator pattern
```

Comment:
`the installation process`
~it required~ `it should be replaced by the operator pattern.`